### PR TITLE
Format Python code of colander to conform to the PEP 8 style guide.

### PR DESCRIPTION
**Pull request by Cykooz**
Python code of ``colander`` formatted to conform to the PEP 8 style guide.
Added some linters (``flake8`` and ``setup.py check``) into ``tox.ini``.

### CODE REVIEW

**Comment by bertjwregeer** [comment]
I would much rather see this go the way that Pyramid/WebOb/Deform have gone and that this gets formatted with `black`.

See:

https://github.com/Pylons/deform/blob/master/tox.ini#L55
https://github.com/Pylons/webob/blob/master/tox.ini#L69
https://github.com/Pylons/pyramid/blob/master/tox.ini#L61

Flake8 configuration:

https://github.com/Pylons/webob/blob/master/.flake8
https://github.com/Pylons/deform/blob/master/.flake8

If necessary or if we want to keep to the 79 characters per line in this project, add a `pyproject.toml`:

https://github.com/Pylons/deform/blob/master/pyproject.toml
https://github.com/Pylons/pyramid/blob/master/pyproject.toml
https://github.com/Pylons/webob/blob/master/pyproject.toml

**Comment by Cykooz** [comment]
> I would much rather see this go the way that Pyramid/WebOb/Deform have gone and that this gets formatted with `black`.

Hm... I has tried apply 'black' to 'colander' code before create this pull request. Its decision about preferred using double quotes instead of single quotes don't like me. I like single quotes.

'black' has an option to enable using single quotes. What do you think about it?

**Comment by mmerickel** [comment]
Please do what @bertjwregeer mentioned and copy Pyramid's config which already has made these decisions (it already preserves single quotes).

**Comment by Cykooz** [comment]
I have finished.

File: .gitignore [hunk 1]
```diff
@@ -23,3 +23,7 @@ coverage.xml
 distribute*.gz
 venv/
 env*/
+.*
+!/.gitignore
+!/.flake8
+!/.travis.yml
```
Comment (bertjwregeer) [inline .gitignore]: Not a huge fan of this pattern, especially since there is another `dot file` that already exists in the repository that is ignored by this policy: `.hgignore`.

Not that it matters much, but I prefer ignoring more specific, vs ignoring all and then whitelisting.

-0 on this change.

File: .travis.yml [hunk 2]
```diff
@@ -24,10 +24,12 @@ matrix:
           env: TOXENV=pypy
         - python: pypy3
           env: TOXENV=pypy3
-        - python: 3.5
-          env: TOXENV=py2-cover,py3-cover,coverage
+        - python: 3.6
+          env: TOXENV=py36-cover,coverage
```
Comment (bertjwregeer) [inline .travis.yml]: Unless you are also dropping Python 2 in this commit, testing needs to be run against Python 2/3 to get full coverage results.

File: tox.ini [hunk 3]
```diff
@@ -1,8 +1,9 @@
 [tox]
 envlist =
+    lint
     py27,py34,py35,py36,py37,py38,pypy,pypy3,
     docs,
-    {py2,py3}-cover,coverage
+    py36-cover,coverage
```
Comment (bertjwregeer) [inline tox.ini]: This goes along with my previous comment, we NEED to run on both python 2 and python 3 to get full coverage results.

Please revert this change.

File: tox.ini [hunk 4]
```diff
@@ -34,26 +35,16 @@ commands =
 # cobertura jenkins reporting and b) pypy and jython can't handle any
 # combination of versions of coverage and nosexcover that i can find.
 
-[py-cover]
+[testenv:py36-cover]
```
Comment (bertjwregeer) [inline tox.ini]: This change too.

File: colander/compat.py [hunk 5]
```diff
@@ -1,38 +1,45 @@
 import sys
 
+
 PY2 = sys.version_info[0] == 2
 PY3 = not PY2
 
-if PY2:
-    string_types = basestring,
+if PY2:  # pragma: no cover
```
Comment (mmerickel) [inline colander/compat.py]: this no cover should not be necessary

File: colander/compat.py [hunk 6]
```diff
@@ -1,38 +1,45 @@
...
 def text_(s, encoding='latin-1', errors='strict'):
     """ If ``s`` is an instance of ``bytes``, return ``s.decode(encoding,
     errors)``, otherwise return ``s``"""
     if isinstance(s, bytes):
         return s.decode(encoding, errors)
-    return s # pragma: no cover
+    return s  # pragma: no cover
+
+
+if PY2:  # pragma: no cover
```
Comment (mmerickel) [inline colander/compat.py]: again, don't add no cover statements

**Comment by mmerickel** [comment]
Awesome, thank you @Cykooz !
